resources/assets/scripts/app/services/download-cached.coffee

```diff
@@ -0,0 +1,36 @@
+# Pony.fm - A community for pony fan music.
+# Copyright (C) 2015 Kelvin Zhang
+#
+# This program is free software: you can redistribute it and/or modify
+# it under the terms of the GNU Affero General Public License as published by
+# the Free Software Foundation, either version 3 of the License, or
+# (at your option) any later version.
+#
+# This program is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+# GNU Affero General Public License for more details.
+#
+# You should have received a copy of the GNU Affero General Public License
+# along with this program.  If not, see <http://www.gnu.org/licenses/>.
+
+angular.module('ponyfm').factory('download-cached', [
+  '$rootScope', '$http', '$log'
+  ($rootScope, $http, $log) ->
+    download = (type, id, format) ->
+      url = '/api/web/' + type + '/cached/' + id + '/' + format
```

Comment:
As this is CoffeeScript, you should use [string interpolation](http://coffeescript.org/#strings) to put a URL like this together.